lib/fluent/plugin/exception_detector.rb

```diff
@@ -41,21 +41,26 @@ def state
       end
     end
 
-    def self.rule(from_state, pattern, to_state)
-      Struct::Rule.new(from_state, pattern, to_state)
+    def self.rule(from_states, pattern, to_state)
+      from_states = [from_states] unless from_states.is_a?(Array)
+      Struct::Rule.new(from_states, pattern, to_state)
     end
 
     def self.supported
       RULES_BY_LANG.keys
     end
 
     JAVA_RULES = [
```

Comment:
[Optional] Yesterday we chat about how we have more strict regex at client side than server side. Have we considered something looser? Like:

```
    JAVA_RULES = [
      rule(:start_state,
           /(?:Exception|Error|Throwable|V8 errors stack trace)[:\r\n]/,
           :java),
      rule(:java, /^[\t ]*nested exception is:[\t ]*$/, :java),
      rule(:java, /^$/, :java),
      rule(:java, /^[\t ]+(?:eval )?at /, :java),
      rule(:java, /^[\t ]*(?:Caused by|Suppressed):/, :java),
      rule(:java, /^[\t ]*... \d+\ more/, :java)
    ].freeze
```

Reply:
That would miss parsing the nested exception itself after the `nested exception is:` line. That's why I've introduced `:java_start_exception`. I suppose we could simply add a transition from `:java` to `:java_start_exception` on `nested exception is:`, but that would not make the rules much simpler. WDYT?